metricflow/plan_conversion/sql_join_builder.py

```diff
@@ -403,64 +412,112 @@ def _make_equality_expression_for_full_outer_join(
-        # meaning the metric time must always be BEFORE the time spine time
+        # meaning the base metrc time must always be BEFORE the comparison metric time
         end_of_range_comparison_expression = SqlComparisonExpression(
-            left_expr=metric_time_column_expr,
+            left_expr=base_column_expr,
             comparison=SqlComparison.LESS_THAN_OR_EQUALS,
-            right_expr=time_spine_column_expr,
+            right_expr=time_comparison_column_expr,
         )
 
         comparison_expressions: List[SqlComparisonExpression] = [end_of_range_comparison_expression]
-        if node.window:
+        if window:
             start_of_range_comparison_expr = SqlComparisonExpression(
-                left_expr=metric_time_column_expr,
+                left_expr=base_column_expr,
                 comparison=SqlComparison.GREATER_THAN,
                 right_expr=SqlSubtractTimeIntervalExpression(
-                    arg=time_spine_column_expr,
-                    count=node.window.count,
-                    granularity=node.window.granularity,
+                    arg=time_comparison_column_expr,
+                    count=window.count,
+                    granularity=window.granularity,
                 ),
             )
             comparison_expressions.append(start_of_range_comparison_expr)
-        elif node.grain_to_date:
+        elif grain_to_date:
             start_of_range_comparison_expr = SqlComparisonExpression(
-                left_expr=metric_time_column_expr,
+                left_expr=base_column_expr,
                 comparison=SqlComparison.GREATER_THAN_OR_EQUALS,
-                right_expr=SqlDateTruncExpression(arg=time_spine_column_expr, time_granularity=node.grain_to_date),
+                right_expr=SqlDateTruncExpression(arg=time_comparison_column_expr, time_granularity=grain_to_date),
             )
             comparison_expressions.append(start_of_range_comparison_expr)
 
-        cumulative_join_condition = SqlLogicalExpression(
+        return SqlLogicalExpression(
             operator=SqlLogicalOperator.AND,
             args=tuple(comparison_expressions),
         )
 
+    @staticmethod
+    def make_join_conversion_join_description(
+        node: JoinConversionEventsNode,
+        base_data_set: AnnotatedSqlDataSet,
+        conversion_data_set: AnnotatedSqlDataSet,
+        column_equality_descriptions: Sequence[ColumnEqualityDescription],
+    ) -> SqlJoinDescription:
+        """Make a join description for joining base-to-conversion data sets.
+
+        A successful conversion is when a conversion event occurs after a corresponding base event
+        under the condition of being within the time range and other conditions (such as constant properties).
+        This builds the join description to satisfy all those conditions.
+        """
+        window_condition = SqlQueryPlanJoinBuilder._make_time_range_window_join_condition(
```

Comment:
I basically just consolidated `_make_time_range_window_join_condition` from the cumulative join logic since they were the exact same with some differences in the beginning. It has some logic around `None` windows. So the logic is essentially this
```
a.ds <= b.ds AND a.ds > b.ds - <window>
If no window is present we join across all time -> "a.ds <= b.ds"
```

Reply:
Right I think the part that confused me was - why do we use `a.ds <= b.ds` for an all time join? 🤔 (I know that's not new logic here though)

Reply:
not all time join, we use a join with a INF backwards looking window. So instead of looking for any base events that converted within an X day window, we just say any base event that converted. Thus, we have `a.ds <= b.ds` which would render out to `base.ds <= conversion.ds`